techniques/applications/repoGpgKeyManagement/1.0/repoGpgKeyManagement.st

```diff
@@ -0,0 +1,135 @@
+#####################################################################################
+#
+#  Copyright (c) 2015 Janos Mattyasovszky
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+bundle agent check_repo_gpg_key
+{
+
+  vars:
+    &GPG_KEY_ACTION:{action |"repo_gpg_action[&i&]" string => "&action&";
+}&
+
+    &GPG_KEY_HASH:{hash |"repo_gpg_hash[&i&]" string => "&hash&";
+}&
+
+    &TRACKINGKEY:{uuid |"repo_gpg_uuid[&i&]" string => "&uuid&";
+}&
+
+    "index_repo_gpg_keys" slist => getindices("repo_gpg_hash");
+
+&GPG_KEY_CONTENT:{content |"repo_gpg_content[&i&]" string => "&content&";
+}&
+
+    ## Unfortunately string_downcase() is only available in 3.6.0, so we need to hack it with some shell
+    "repo_gpg_hash_lc[${index_repo_gpg_keys}]" string => execresult("/bin/echo ${repo_gpg_hash[${index_repo_gpg_keys}]} | tr \"[:upper:]\" \"[:lower:]\"","useshell");
+
+   debian|ubuntu::
+    "repo_gpg_hash_uc[${index_repo_gpg_keys}]" string => execresult("/bin/echo ${repo_gpg_hash[${index_repo_gpg_keys}]} | tr \"[:lower:]\" \"[:upper:]\"","useshell");
+
+
+  classes:
+    "repo_gpg_key_${index_repo_gpg_keys}_action_remove"
+      expression => strcmp("${repo_gpg_action[${index_repo_gpg_keys}]}","del");
+
+    "repo_gpg_key_${index_repo_gpg_keys}_action_add"
+      expression => strcmp("${repo_gpg_action[${index_repo_gpg_keys}]}","add");
+
+  files:
+    "${g.rudder_var_tmp}/gpg_key_${index_repo_gpg_keys}_content"
+      comment       => "Create a temporary file for repo to import the GPG key. Unfortunately rpm does not accept /dev/stdin or - for importing",
+      create        => "true",
+      edit_line     => set_gpg_file_content( "${repo_gpg_content[${index_repo_gpg_keys}]}" ),
+      edit_defaults => empty_nobackup,
+      ifvarclass    => "!repo_gpg_key_${repo_gpg_hash_lc[${index_repo_gpg_keys}]}_present.repo_gpg_key_${index_repo_gpg_keys}_action_add",
+      classes       => if_else("gpg_key_file_${index_repo_gpg_keys}_created", "gpg_key_file_${index_repo_gpg_keys}_create_failed");
```

Comment:
you also may want to consider using the classes_generic rather than if_else (if gives also the _reached)
It makes the promises easier to read (but the report a bit hard to understand though..)